react-router/history.d.ts

```diff
@@ -0,0 +1,183 @@
+// Type definitions for history v1.11.1
+// Project: https://github.com/rackt/history
+// Definitions by: Sergey Buturlakin <http://github.com/sergey-buturlakin>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+
+declare namespace HistoryModule {
+
+    // types based on https://github.com/rackt/history/blob/master/docs/Terms.md
+
+    type Action = string
+
+    type BeforeUnloadHook = () => string
+
+    type CreateHistory<T> = (options?: HistoryOptions) => T
+
+    type CreateHistoryEnhancer<T, E> = (createHistory: CreateHistory<T>) => CreateHistory<T & E>
+
+    interface History {
+        listenBefore(hook: TransitionHook): Function
+        listen(listener: LocationListener): Function
+        transitionTo(location: Location): void
+        pushState(state: LocationState, path: Path): void
+        replaceState(state: LocationState, path: Path): void
```

Comment:
They accept, but in case when you apply history enhancer function to a history object:
let history = useQueries(createHistory)
// this history should have updated pushState function with query parameter
history.pushState(state: LocationState, path: Path, query?: Query): void